# Day 2: Colab 2C - Five Claims About a Dataset You've Never Seen

Names: Calvin and Ethan

Date: 09/01/2026

In [5]:
import pandas as pd

url = 'https://eds-217-essential-python.github.io/data/national_parks.csv'

df = pd.read_csv(url)

In [8]:
df.shape

(21560, 12)

In [10]:
df.columns.tolist()

['year',
 'gnis_id',
 'geometry',
 'metadata',
 'number_of_records',
 'parkname',
 'region',
 'state',
 'unit_code',
 'unit_name',
 'unit_type',
 'visitors']

There are 12 columns and they are called year', 'gnis_id', 'geometry', 'metadata', 'number_of_records', 'parkname', 'region', 'state', 'unit_code', 'unit_name', 'unit_type', 'visitors'

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21560 entries, 0 to 21559
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   year               21560 non-null  object 
 1   gnis_id            21560 non-null  object 
 2   geometry           21560 non-null  object 
 3   metadata           18848 non-null  object 
 4   number_of_records  21560 non-null  int64  
 5   parkname           19342 non-null  object 
 6   region             21560 non-null  object 
 7   state              21560 non-null  object 
 8   unit_code          21560 non-null  object 
 9   unit_name          21560 non-null  object 
 10  unit_type          21560 non-null  object 
 11  visitors           21556 non-null  float64
dtypes: float64(1), int64(1), object(10)
memory usage: 2.0+ MB


In [ ]:
df.dtypes

type(df.dtypes)

pandas.core.series.Series

It says year is an object but it should int

It seems these individual columns are strings 

In [13]:
df['year']

0        1904
1        1941
2        1961
3        1935
4        1982
         ... 
21555    1905
21556    1905
21557    1905
21558    1905
21559    1905
Name: year, Length: 21560, dtype: object

In [18]:
df.info

<bound method DataFrame.info of        year  gnis_id      geometry  \
0      1904  1163670       POLYGON   
1      1941  1531834  MULTIPOLYGON   
2      1961  2055170  MULTIPOLYGON   
3      1935  1530459  MULTIPOLYGON   
4      1982   277263       POLYGON   
...     ...      ...           ...   
21555  1905  1163670       POLYGON   
21556  1905  1609331       POLYGON   
21557  1905   255948  MULTIPOLYGON   
21558  1905  1528416  MULTIPOLYGON   
21559  1905  1263056  MULTIPOLYGON   

                                                metadata  number_of_records  \
0                                                    NaN                  1   
1                                                    NaN                  1   
2                                                    NaN                  1   
3                                                    NaN                  1   
4                                                    NaN                  1   
...                                  

In [19]:
df.isnull().sum()

year                    0
gnis_id                 0
geometry                0
metadata             2712
number_of_records       0
parkname             2218
region                  0
state                   0
unit_code               0
unit_name               0
unit_type               0
visitors                4
dtype: int64

metadata and parkname are both missing values. meadata is missing 2712 and parkname is missing 2218.

In [34]:

df['year'].value_counts
df['gnis_id'].value_counts
df['visitors'].value_counts

<bound method IndexOpsMixin.value_counts of 0          1500.0
1             0.0
2         69000.0
3          2200.0
4        468144.0
           ...   
21555      1400.0
21556     26188.0
21557      1000.0
21558       928.0
21559      2438.0
Name: visitors, Length: 21560, dtype: float64>

In [35]:
df.describe()

,number_of_records,visitors
count,21560.0,2.155600e+04
mean,1.0,1.277105e+06
std,0.0,1.190148e+07
min,1.0,0.000000e+00
25%,1.0,3.912475e+04
50%,1.0,1.552190e+05
75%,1.0,6.081438e+05
max,1.0,8.719228e+08


The visitors column's mean is very far from the median. The data is very skewed. 

In [46]:
keepcol = [
    'year', 'gnis_id', 'visitors'
]

df_new = df[keepcol]

In [50]:
rename_col = {
    'year' : 'year',
    'gnis_id' : 'id',
    'visitors' : 'visitor'
}

df_newer = df_new.rename(columns = rename_col)

In [57]:
df["visitors"].value_counts()

visitors
0.0          180
2500.0        16
500.0         15
1000.0        14
5000.0        14
            ... 
54988.0        1
9514.0         1
23465.0        1
2432348.0      1
44182.0        1
Name: count, Length: 18915, dtype: int64

In [59]:
df["parkname"].value_counts()

parkname
Denali              192
Great Sand Dunes    170
Katmai              126
Glacier Bay         124
Yellowstone         114
                   ... 
Keweenaw              2
Honouliuli            1
Pullman               1
Valles Caldera        1
Minidoka              1
Name: count, Length: 336, dtype: int64

# 5 Claims

##1 Out of all of the park visits, there were 180 recordings where there were no visitors on a given day

##2 Based on the .value_counts() function on the original df, Denali has 192 recordings, which is 22 more than the Great Sand Dunes. 

##3 The visitor column is very skewed because the median is 155,219 and the mean is 1,277,104. This proves that there is something wrong with the data and some aspects were not consistently recorded. 


##4 The data format for year is wrong because it comes back as a list of strings when it should be a series of integers.


##5 The park names column is missing data because there are null values. This will make it very difficult to run the analysis since over 2,000 rows do not have a park associated with the data.